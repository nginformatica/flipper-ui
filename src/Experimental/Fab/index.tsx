export { default as Fab } from './Fab'
export { default as FabWrapper } from './FabWrapper'

// export more components, hooks or helpers below if needed
