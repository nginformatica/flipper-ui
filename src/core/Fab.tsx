import { Fab as MuiFab } from '@material-ui/core'
import React, { MouseEvent } from 'react'
import { DefaultProps } from './types'

interface FabProps extends DefaultProps {
    disabled?: boolean
    component?: string
    color?: 'default' | 'primary' | 'inherit' | 'secondary'
    size?: 'small' | 'medium' | 'large'
    href?: string
    variant?: 'round' | 'extended'
    onClick?: (event?: MouseEvent<HTMLElement>) => void
    children: React.ReactElement<{}>
}

const Fab = ({
    children,
    margin,
    padding,
    style = {},
    variant,
    ...otherProps
}: FabProps) => (
    <MuiFab
        {...otherProps}
        variant={variant}
        style={{ margin, padding, ...style }}>
        {children}
    </MuiFab>
)

export default Fab
