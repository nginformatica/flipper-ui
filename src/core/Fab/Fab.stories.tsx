import React from 'react'
import { Meta, StoryFn } from '@storybook/react'
import Fab from '.'

export default {
    title: 'Fab',
    component: Fab
} as Meta<typeof Fab>

const Template: StoryFn<typeof Fab> = args => <Fab {...args} />
export const Default = Template.bind({})

Default.args = {
    children: 'Fab'
}
