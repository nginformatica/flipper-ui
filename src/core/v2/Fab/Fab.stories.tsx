import React from 'react'
import FabV2 from './Fab'
import FabWrapper from './FabWrapper'
import FileCopyIcon from '@material-ui/icons/FileCopy'

export const fab = () => (
    <FabV2 name='copy' tooltip='Copy file' onClick={() => alert('fab click')}>
        {<FileCopyIcon />}
    </FabV2>
)

export default {
    title: 'V2/Fab',
    component: FabV2,
    subcomponents: {
        FabWrapper
    }
}
