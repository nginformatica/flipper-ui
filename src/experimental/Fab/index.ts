export { default as Fab } from './Fab'
export { default as FabWrapper } from './FabWrapper'
