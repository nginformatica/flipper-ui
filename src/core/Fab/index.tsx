import { Fab as MuiFab } from '@material-ui/core'
import React, { MouseEvent } from 'react'
import { DefaultProps } from '../types'

export interface FabProps extends DefaultProps {
    disabled?: boolean
    component?: string
    color?: 'default' | 'primary' | 'inherit' | 'secondary'
    size?: 'small' | 'medium' | 'large'
    href?: string
    variant?: 'round' | 'extended'
    onClick?: (event?: MouseEvent<HTMLElement>) => void
    children: React.ReactElement<Record<string, unknown>>
}

const Fab = ({
    children,
    margin,
    padding,
    style = {},
    variant,
    ...otherProps
}: FabProps) => (
    <MuiFab
        {...otherProps}
        variant={variant}
        style={{ margin, padding, ...style }}>
        {children}
    </MuiFab>
)

export default Fab
