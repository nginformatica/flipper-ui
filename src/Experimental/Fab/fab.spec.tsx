/* eslint-disable max-len */
import * as React from 'react'
import { render, screen, waitFor } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import Fab from './Fab'
import FabWrapper from './FabWrapper'
import FileCopyIcon from '@material-ui/icons/FileCopy'

describe('Fab', () => {
    it('should render correctly', () => {
        render(
            <Fab onClick={jest.fn()}>
                <FileCopyIcon />
            </Fab>
        )

        const container = screen.getByTestId('fab-button')

        expect(container.classList).toContain('MuiFab-sizeSmall')
    })

    it('should render large correctly', () => {
        render(
            <Fab large onClick={jest.fn()}>
                <FileCopyIcon />
            </Fab>
        )

        const container = screen.getByTestId('fab-button')

        expect(container.classList).toContain('MuiFab-sizeMedium')
    })

    it('Should call onClick', () => {
        const onClickSpy = jest.fn()
        render(
            <Fab onClick={onClickSpy}>
                <FileCopyIcon />
            </Fab>
        )

        const btn = screen.getByTestId('fab-button')

        userEvent.click(btn)

        waitFor(() => {
            expect(onClickSpy).toHaveBeenCalled()
        })
    })

    it('Should show tooltip', () => {
        const onClickSpy = jest.fn()
        render(
            <Fab onClick={onClickSpy} tooltip='tooltip-test'>
                <FileCopyIcon />
            </Fab>
        )

        const btn = screen.getByTestId('fab-button')

        userEvent.hover(btn)

        waitFor(() => {
            const tooltipMessage = screen.getByText('tooltip-test')
            expect(tooltipMessage).toBeDefined()
        })
    })
})

describe('FabWrapper', () => {
    it('should render correctly', () => {
        render(<FabWrapper>Test</FabWrapper>)

        const fabWrapper = screen.getByText('Test')

        expect(fabWrapper).toBeDefined()
    })
})
