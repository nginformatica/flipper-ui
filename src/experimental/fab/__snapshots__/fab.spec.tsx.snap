// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fab should match snapshot 1`] = `
<div>
  <div
    class=""
    style="display: inherit;"
    title="tooltip-test"
  >
    <button
      class="MuiButtonBase-root MuiFab-root MuiFab-sizeSmall"
      data-testid="fab-button"
      name="action-button"
      tabindex="0"
      tooltip="tooltip-test"
      type="button"
    >
      <span
        class="MuiFab-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="FileCopyIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12zm-1 4 6 6v10c0 1.1-.9 2-2 2H7.99C6.89 23 6 22.1 6 21l.01-14c0-1.1.89-2 1.99-2zm-1 7h5.5L14 6.5z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`FabWrapper should match snapshot 1`] = `
<div>
  <div
    class="sc-fBWQee lhYxDZ list-item-action"
  >
    Test
  </div>
</div>
`;
